src/chainparams.cpp

```diff
@@ -90,11 +86,17 @@ class CMainParams : public CChainParams {
         vSeeds.push_back(CDNSSeedData("xf2.org", "bitseed.xf2.org")); // Jeff Garzik
         vSeeds.push_back(CDNSSeedData("bitcoin.jonasschnelli.ch", "seed.bitcoin.jonasschnelli.ch")); // Jonas Schnelli
 
-        base58Prefixes[PUBKEY_ADDRESS] = std::vector<unsigned char>(1,0);
-        base58Prefixes[SCRIPT_ADDRESS] = std::vector<unsigned char>(1,5);
-        base58Prefixes[SECRET_KEY] =     std::vector<unsigned char>(1,128);
-        base58Prefixes[EXT_PUBLIC_KEY] = boost::assign::list_of(0x04)(0x88)(0xB2)(0x1E).convert_to_container<std::vector<unsigned char> >();
-        base58Prefixes[EXT_SECRET_KEY] = boost::assign::list_of(0x04)(0x88)(0xAD)(0xE4).convert_to_container<std::vector<unsigned char> >();
+        const unsigned char chPubkeyAddress[] = { 0 };
+        const unsigned char chScriptAddress[] = { 5 };
+        const unsigned char chSecretKey[] =     { 128 };
+        const unsigned char chExtPublicKey[] =  { 0x04, 0x88, 0xB2, 0x1E };
+        const unsigned char chExtSecretKey[] =  { 0x04, 0x88, 0xAD, 0xE4 };
```

Comment:
I was also removing boost dependencies from consensus code, but now I am not sure what is preferred. Chainparams is out of consensus and it seems reasonable to move towards c++11.
But maybe we want to eventually have a libconsensus not only exposed but also written in C?
In any case, I would really like to have a more clear criterion.
My idea was "remove boost when possible", then @laanwj told be that boost/foreach was there for later c++11, and I thought that was an exception.
But now I'm really confused and I definitely want to avoid going in circles.